src/v/cloud_storage/inventory/inv_ops.cc

```diff
@@ -29,49 +29,50 @@ namespace cloud_storage::inventory {
 inv_ops::inv_ops(ops_t ops)
   : _inv_ops{std::move(ops)} {}
 
-ss::future<cloud_storage::upload_result>
-inv_ops::create_inventory_configuration(
+ss::future<op_result<void>> inv_ops::create_inventory_configuration(
   cloud_storage_api& remote, retry_chain_node& parent) {
     return ss::visit(_inv_ops, [&remote, &parent](auto& ops) {
         return ops.create_inventory_configuration(
           remote, parent, frequency, format);
     });
 }
 
-ss::future<bool> inv_ops::inventory_configuration_exists(
+ss::future<op_result<bool>> inv_ops::inventory_configuration_exists(
   cloud_storage_api& remote, retry_chain_node& parent) {
     return ss::visit(_inv_ops, [&remote, &parent](auto& ops) {
         return ops.inventory_configuration_exists(remote, parent);
     });
 }
 
-ss::future<inventory_creation_result>
+ss::future<op_result<inventory_creation_result>>
 inv_ops::maybe_create_inventory_configuration(
   cloud_storage_api& remote, retry_chain_node& parent) {
     if (const auto exists = co_await inventory_configuration_exists(
           remote, parent);
-        exists) {
+        exists.has_value() && exists.value()) {
         co_return inventory_creation_result::already_exists;
     }
```

Comment:
If the result contains an error, should we propagate that error up instead of creating a new one?

Reply:
In this specific line there is no error as the config exists, so we return a success value. In general though, in earlier PRs we propagated the http client errors up to the caller, I initially thought of somehow mapping http client errors to values in the new `error_outcome` enum that we are using for inventory module, but I decided against it as the caller does not care about the http error type in this case, just that something has failed. 

We need logging of the underlying error here which is missing right now, once logs are added here (WIP) I think this should be sufficient for troubleshooting. 

However if you think returning the http client errors is clearer then I can give mapping the error outcomes another try.

Reply:
I don't think progragating the exact http error is necessarily important (especially if you're adding some logging), but at least it seems like `inventory_configuration_exists()` can return `failed_to_check_inv_status` which seems like it should be bubbled up? As is it looks like we proceed to creation. I suppose it's likely that whatever error was seen in checking existence will also be seen below, but still seems like surprising behavior

Reply:
I see what you mean now. Yeah that needs to be fixed, will change this now. Thanks for pointing out.

Reply:
changed the structure and added error logging